lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4757,6 +4757,17 @@ Instruction *SPIRVToLLVM::transOCLBuiltinFromExtInst(SPIRVExtInst *BC,
          "Not OpenCL extended instruction");
 
   std::vector<Type *> ArgTypes = transTypeVector(BC->getArgTypes(), true);
+  for (unsigned I = 0; I < ArgTypes.size(); I++) {
```

Comment:
Does function transBuiltinFromInst() also need this special handling?  It also calls transTypeVector() and gets a mangled name.

Reply:
My experiments show that e.g. for `OpEnqueKernel` we don't meet untyped variables in arguments, but the mangling is still wrong - should be fixed elsewhere.
It's still possible to unify this fix inside `transTypeVector()`, but not sure if it'd be used by `transBuiltinFromInst()`.

Reply:
I'm fine with the merge as is for now. In general we probably should implement the following algorithm:
- translate SPIR-V pointer type as LLVM's Typed pointer
- get the mangling
- 'opaquefy' the pointer to construct the function and call instruction